zrml/prediction-markets/src/tests.rs

```diff
@@ -1476,6 +1476,404 @@ fn on_market_close_successfully_auto_closes_multiple_markets_after_stall() {
+        let market_after_close = MarketCommons::market(&0).unwrap();
+        assert_eq!(market_after_close.status, MarketStatus::Active);
+        let pool_after_close = Swaps::pool(0).unwrap();
+        assert_eq!(pool_after_close.pool_status, PoolStatus::Active);
+
+        let market_after_close = MarketCommons::market(&1).unwrap();
+        assert_eq!(market_after_close.status, MarketStatus::Active);
+        let pool_after_close = Swaps::pool(1).unwrap();
+        assert_eq!(pool_after_close.pool_status, PoolStatus::Active);
+    });
+}
+
+#[test]
+fn manually_open_market_after_long_stall() {
+    ExtBuilder::default().build().execute_with(|| {
+        // Mock last time frame to prevent it from defaulting.
+        LastTimeFrame::<Runtime>::set(Some(0));
+
+        let start: Moment = (33 * MILLISECS_PER_BLOCK).into();
+        let end: Moment = (666 * MILLISECS_PER_BLOCK).into();
+        let category_count = 3;
+        assert_ok!(PredictionMarkets::create_cpmm_market_and_deploy_assets(
+            RuntimeOrigin::signed(ALICE),
+            Asset::Ztg,
+            Perbill::zero(),
+            ALICE,
+            MarketPeriod::Timestamp(start..end),
+            get_deadlines(),
+            gen_metadata(50),
+            MarketType::Categorical(category_count),
+            Some(MarketDisputeMechanism::Court),
+            0,
+            LIQUIDITY,
+            vec![<Runtime as zrml_swaps::Config>::MinWeight::get(); category_count.into()],
+        ));
+        assert_ok!(PredictionMarkets::create_cpmm_market_and_deploy_assets(
+            RuntimeOrigin::signed(ALICE),
+            Asset::Ztg,
+            Perbill::zero(),
+            ALICE,
+            MarketPeriod::Timestamp(start..end),
+            get_deadlines(),
+            gen_metadata(50),
+            MarketType::Categorical(category_count),
+            Some(MarketDisputeMechanism::Court),
+            0,
+            LIQUIDITY,
+            vec![<Runtime as zrml_swaps::Config>::MinWeight::get(); category_count.into()],
+        ));
+
+        set_timestamp_for_on_initialize(
+            start + (crate::MAX_RECOVERY_TIME_FRAMES + 1) * MILLISECS_PER_BLOCK as u64,
+        );
+        run_to_block(2); // Trigger `on_initialize`; must be at least block #2!
+
+        // Pools were not opened, but are still initialized, because recovery limit reached.
+        assert_eq!(Swaps::pool(0).unwrap().pool_status, PoolStatus::Initialized);
+        assert_eq!(Swaps::pool(1).unwrap().pool_status, PoolStatus::Initialized);
+
+        assert_ok!(PredictionMarkets::manually_open_or_close_market(
+            RuntimeOrigin::signed(ALICE),
+            0
+        ));
+        assert_eq!(Swaps::pool(0).unwrap().pool_status, PoolStatus::Active);
+
+        assert_ok!(PredictionMarkets::manually_open_or_close_market(
+            RuntimeOrigin::signed(ALICE),
+            1
+        ));
+        assert_eq!(Swaps::pool(1).unwrap().pool_status, PoolStatus::Active);
```

Comment:
Can we add checks here that the IDs are properly cleared from the auto-open vectors?

Reply:
7012b50cdebd68f9e558d92fe689058e54013343